pkg/controller/pcideviceclaim/pcideviceclaim_controller.go

```diff
@@ -331,49 +286,144 @@ func (h Handler) rebindAfterReboot() error {
 
 func (h Handler) reconcilePCIDeviceClaims(name string, pdc *v1beta1.PCIDeviceClaim) (*v1beta1.PCIDeviceClaim, error) {
 
-	if pdc == nil || pdc.DeletionTimestamp != nil {
+	if pdc == nil || pdc.DeletionTimestamp != nil || (pdc.Spec.NodeName != h.nodeName) {
 		return pdc, nil
 	}
 
-	if pdc.Spec.NodeName == h.nodeName && !pdc.Status.PassthroughEnabled {
-		newPdc, err := h.attemptToEnablePassthrough(pdc)
-		return newPdc, err
+	// Get the PCIDevice object for the PCIDeviceClaim
+	pd, err := h.getPCIDeviceForClaim(pdc)
+	if pd == nil {
+		return pdc, err
+	}
+
+	// Find the DevicePlugin
+	resourceName := pd.Status.ResourceName
+	dp := deviceplugins.Find(
+		resourceName,
+		h.devicePlugins,
+	)
+
+	h.permitHostDeviceInKubeVirt(pd)
+
+	// If passthrough is already enabled and there's a deviceplugin, then we can return
+	if pdc.Status.PassthroughEnabled && (dp != nil){
+		return pdc, nil
 	}
 
-	return pdc, nil
+	// Enable PCI Passthrough on the device by binding it to vfio-pci driver
+	newPdc, err := h.attemptToEnablePassthrough(pd, pdc)
```

Comment:
```suggestion
	if err := h.permitHostDeviceInKubeVirt(pd); err != nil {
		return pdc, fmt.Errorf("error updating kubevirt CR: %v", err)
	}
```
added error handling to ensure this is retried in case of issues.